File: gerenciapet-frontend/src/app/models/Aluno.ts
import User from "./User";
import Inscricao from "./Inscricao";

export default class Aluno extends User {
    public matricula: string = '';
    public inscricoes: Inscricao[] = [];

    constructor(object?: any) {
        super(object);
        if (object !== undefined) {
            this.updateAluno(object);
        }
    }

    public updateAluno(object: any) {
        this.matricula = object.matricula ?? '';
        if (object.inscricoes) {
            this.inscricoes = object.inscricoes;
        }
    }
}

File: gerenciapet-backend/src/main/java/br/ufes/gerenciapet/backend/model/Aluno.java
package br.ufes.gerenciapet.backend.model;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.OneToMany;
import java.util.ArrayList;
import java.util.List;

@Entity
public class Aluno extends User {

    @Column(nullable = true, length = 20)
    private String matricula;

    @OneToMany(mappedBy = "aluno")
    @com.fasterxml.jackson.annotation.JsonIgnore
    private List<Inscricao> inscricoes = new ArrayList<>();

    public Aluno() {}

    public void inscreverEmEvento(Evento evento) {
        // Implementação futura
    }

    public void baixarCertificado(Evento evento) {
        // Implementação futura
    }

    public String getMatricula() { return matricula; }
    public void setMatricula(String matricula) { this.matricula = matricula; }
    public List<Inscricao> getInscricoes() { return inscricoes; }
    public void setInscricoes(List<Inscricao> inscricoes) { this.inscricoes = inscricoes; }
}
